src/richie/apps/core/context_processors.py

```diff
@@ -33,46 +34,65 @@ def site_metas(request):
+                "release": getattr(settings, "RELEASE", ""),
+                "sentry_dsn": getattr(settings, "SENTRY_DSN", ""),
+            }
         },
-        "FRONTEND_CONTEXT": json.dumps(
+    }
+
+    if getattr(settings, "CDN_DOMAIN", False):
+        context["CDN_DOMAIN"] = settings.CDN_DOMAIN
+
+    if getattr(settings, "AUTHENTICATION_DELEGATION", False):
+        context.update(
             {
-                "context": {
-                    "csrftoken": get_token(request),
-                    "environment": getattr(settings, "ENVIRONMENT", ""),
-                    "release": getattr(settings, "RELEASE", ""),
-                    "sentry_dsn": getattr(settings, "SENTRY_DSN", ""),
-                    "authentication": {
-                        "endpoint": authentication_delegation["BASE_URL"],
-                        "backend": authentication_delegation["BACKEND"],
-                    },
-                    "lms_backends": [
-                        {
-                            "endpoint": lms["BASE_URL"],
-                            "backend": lms["BACKEND"],
-                            "course_regexp": lms["JS_COURSE_REGEX"],
-                            "selector_regexp": lms["JS_SELECTOR_REGEX"],
-                        }
-                        for lms in getattr(settings, "LMS_BACKENDS", [])
-                    ],
+                "AUTHENTICATION": {
+                    "PROFILE_URLS": json.dumps(
+                        [
+                            {
+                                "label": str(url["label"]),
+                                "action": str(
+                                    url["href"].format(
+                                        base_url=authentication_delegation["BASE_URL"]
+                                    )
+                                ),
+                            }
+                            for url in getattr(
+                                authentication_delegation, "PROFILE_URLS", []
+                            )
+                        ]
+                    ),
                 }
             }
+        )
+
+        context["FRONTEND_CONTEXT"]["context"].update(
+            {
+                "authentication": {
+                    "endpoint": authentication_delegation["BASE_URL"],
+                    "backend": authentication_delegation["BACKEND"],
+                }
+            }
+        )
+
+    if getattr(settings, "LMS_BACKENDS", False):
+        context["FRONTEND_CONTEXT"]["context"].update(  # NOQA
+            {
+                "lms_backends": [
+                    {
+                        "endpoint": lms["BASE_URL"],
+                        "backend": lms["BACKEND"],
+                        "course_regexp": lms["JS_COURSE_REGEX"],
+                        "selector_regexp": lms["JS_SELECTOR_REGEX"],
+                    }
+                    for lms in getattr(settings, "LMS_BACKENDS", [])
+                ]
+            }
```

Comment:
```suggestion
        context["FRONTEND_CONTEXT"]["context"]["lms_backends"] = [
            {
                "endpoint": lms["BASE_URL"],
                "backend": lms["BACKEND"],
                "course_regexp": lms["JS_COURSE_REGEX"],
                "selector_regexp": lms["JS_SELECTOR_REGEX"],
            }
            for lms in getattr(settings, "LMS_BACKENDS", [])
        ]
```

Reply:
If I use this syntax, the `json_dumps` produces a bad result 😕

I have `["lms_backends"] = [[{ endpoint: ... }]]` instead of `[{ endpoint: ... }]`.

Reply:
hmm I don't see how it is possible. I think it might be that you had a "," between the "}" and the "for"?

Reply:
I'll work on it. But I have exactly the same issue to set `context["FRONTEND_CONTEXT"]["context"]["authentication"]`. If I do not use `.update()`, authentication properties are wrapped in an array.

Reply:
🍷